CMakeLists.txt

```diff
@@ -285,6 +286,19 @@ if(BUILD_PYTHON)
     target_link_libraries(retro retro-base ${PYBIND_LIBS} ${STATIC_LDFLAGS})
 endif()
 
+if(BUILD_C)
+    add_library(retro-c SHARED src/retro-c.cpp)
+    set_target_properties(retro-c PROPERTIES 
+        OUTPUT_NAME retro
+        CXX_VISIBILITY_PRESET default)
+    if(WIN32)
+        if(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "AMD64")
+            add_definitions(-DMS_WIN64)
```

Comment:
Why is this needed here? We already have add_definitions for if you're building the Python bindings. That should probably be pulled out and shared.